src/dbnode/encoding/m3tsz/encoder.go

```diff
@@ -116,6 +91,10 @@ func (enc *encoder) Encode(dp ts.Datapoint, tu xtime.Unit, ant ts.Annotation) er
 	}
 
 	var err error
+	if err = enc.tsEncoderState.WriteTime(enc.os, dp.Timestamp, ant, tu); err != nil {
```

Comment:
nit: After this was just refactored, maybe it would be better to just structure as:
```go
err := enc.tsEncoderState.WriteTime(enc.os, dp.Timestamp, ant, tu)
if err != nil {
  return err
}
```